src/main/java/me/gaminglounge/guiapi/Pagenation.java

```diff
@@ -3,101 +3,153 @@
+     * - setItems()
+     */
+    public Pagenation(Player p) {
+        this.player = p;
         mm = MiniMessage.miniMessage();
-    }
 
-    public Pagenation setPlaceholderItem(ItemStack placeholderItem) {
-        this.placeholderItem = placeholderItem;
-        return this;
-    }
+        ItemBuilderManager.addBothClickEvent("GuiAPI:nextPage", (e) -> {
+            if (e.getInventory().getHolder() instanceof Pagenation pgi) {
+                e.setCancelled(true);
+                int targetPage = pgi.currentpage + 1;
+                if (targetPage < 0 || targetPage > pgi.items.size() / Pagenation.numItemsOnPage) {
+                    // error
+                    return;
+                }
+                pgi.fillPage(targetPage);
+            }
+        });
+
+        ItemBuilderManager.addBothClickEvent("GuiAPI:prevPage", (e) -> {
+            if (e.getInventory().getHolder() instanceof Pagenation pgi) {
+                e.setCancelled(true);
+                int targetPage = pgi.currentpage - 1;
+                if (targetPage < 0 || targetPage > (pgi.items.size() + Pagenation.numItemsOnPage - 1)
+                        / Pagenation.numItemsOnPage) {
+                    // error
+                    return;
+                }
+                pgi.fillPage(targetPage);
+            }
+        });
+
+        ItemBuilderManager.addBothClickEvent("GuiAPI:pagenation_return_to_old_inv_or_close", (event) -> {
+            event.setCancelled(true);
+            if (event.getInventory().getHolder() instanceof Pagenation pn) {
+                if (pn.returnInventory == null) {
+                    event.getWhoClicked().closeInventory();
+                } else {
+                    event.getWhoClicked().openInventory(pn.returnInventory);
+                }
+            }
+        });
 
-    public Pagenation setplaceholderItemIndex(int[] placeholderItemIndex) {
-        this.placeholderItemIndex = placeholderItemIndex;
-        return this;
     }
 
+    /*
+     * Sets the inventory name
+     */
     public Pagenation setInventoryName(Component inventoryName) {
         this.inventoryName = inventoryName;
         return this;
     }
-    
+
+    /*
+     * @param items - List of items to be displayed in the pagenation
+     */
     public Pagenation setItems(List<ItemStack> items) {
         this.items = items;
         return this;
     }
 
-    public Pagenation setBack(ItemStack back) {
-        this.back = back;
+    /*
+     * 
+     */
+    public Pagenation setBackInv(Inventory returnInventory) {
+        this.returnInventory = returnInventory;
         return this;
     }
 
-    public Pagenation setNext(ItemStack next) {
-        this.next = next;
-        return this;
-    }
-
-    public Pagenation setClose(ItemStack close) {
-        this.close = close;
-        return this;
-    }
-
-
+    /*
+     * @return the inventory
+     */
     @Override
     public @NotNull Inventory getInventory() {
```

Comment:
Javadoc comments should be marked as such.
```suggestion
    /**
     * @return the inventory
     */
    @Override
    public @NotNull Inventory getInventory() {
```